parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -169,13 +169,23 @@ public void initialize(MessageType fileSchema,
     // initialize a ReadContext for this file
     ReadSupport.ReadContext readContext = readSupport.init(new InitContext(
         configuration, toSetMultiMap(fileMetadata), fileSchema));
-    this.requestedSchema = readContext.getRequestedSchema();
+    MessageType request = readContext.getRequestedSchema();
+    boolean isCaseSensitive = configuration.getBoolean(ParquetInputFormat.CASE_SENSITIVITY, true);
+
+    if (!isCaseSensitive) {
+      MessageType base = new MessageType(request, false);
+      this.requestedSchema = new MessageType(request.getName(), fileSchema.getContainedFields(base));
```

Comment:
instead of adding isCaseSensitive in base, it can be passed to getContainedField